fmpz/multi_crt.c

```diff
@@ -442,33 +533,13 @@ void _fmpz_multi_crt_run_p(
     const fmpz * const * inputs)
 {
     slong i;
-    slong a, b, c;
-    const fmpz * B, * C;
-    fmpz * A, * t1, * t2;
+    fmpz * ins = FLINT_ARRAY_ALLOC(P->moduli_count, fmpz);
 
-    t1 = outputs + P->temp1loc;
-    t2 = outputs + P->temp2loc;
-
-    for (i = 0; i < P->length; i++)
-    {
-        a = P->prog[i].a_idx;
-        b = P->prog[i].b_idx;
-        c = P->prog[i].c_idx;
-        FLINT_ASSERT(a >= 0);
-        A = outputs + a;
-        B = b < 0 ? inputs[-b-1] : outputs + b;
-        C = c < 0 ? inputs[-c-1] : outputs + c;
+    for (i = 0; i < P->moduli_count; i++)
+        ins[i] = *inputs[i];
 
-        /* A = B + I*(C - B) mod M */
-        fmpz_sub(t1, B, C);
-        fmpz_mul(t2, P->prog[i].idem, t1);
-        fmpz_sub(t1, B, t2);
-        fmpz_smod(A, t1, P->prog[i].modulus);
+    _fmpz_multi_CRT_run(outputs, P, ins, 1);
 
-        /* last calculation should write answer to outputs[0] */
-        if (i + 1 >= P->length)
-        {
-            FLINT_ASSERT(A == outputs + 0);
-        }
-    }
+    flint_free(ins);
```

Comment:
How does this clean up the fmpz's?

Reply:
It doesn't need to as it never created any.